### PR TITLE
stm32mp1: u-boot fragment and StMM support

### PR DESCRIPTION
**Pull request by etienne-lms**
Add U-Boot configuration fragments support for stm32mp1 platforms.

Add StMM support for stm32mp157*-ed1/-ev1 platforms upon added build directive WITH_STMM.
StMM build is currently broken hence this P-R depends on https://github.com/OP-TEE/manifest/pull/282 to be fully functional.

### CODE REVIEW

File: stm32mp1.mk [hunk 1]
```diff
@@ -70,6 +70,13 @@ STM32MP1_DTS_LINUX ?= $(STM32MP1_DTS_BASENAME)
 STM32MP1_DTS_U_BOOT ?= $(STM32MP1_DTS_BASENAME)
 STM32MP1_DEFCONFIG_U_BOOT ?= stm32mp15_defconfig
 
+# When enabled, WITH_STMM embeds StMM application in OP-TEE OS and default
+# enables WITH_RPMB_TEST for RPMB secure storage which StMM relies on.
+WITH_STMM ?= n
+ifeq ($(WITH_STMM),y)
+WITH_RPMB_TEST ?= y
```
Comment (jenswi-linaro) [inline stm32mp1.mk:77]: How about `WITH_RPMB_TEST ?= $(WITH_STMM)`? Doesn't that do the same thing as these three lines?

Comment (etienne-lms) [reply stm32mp1.mk:77]: Right. I'll update.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Tag applied.
This depends on OP-TEE/manifest#282. @jenswi-linaro, would you mind having a look at it?

**Comment by jenswi-linaro** [comment]
@etienne-lms, sure.

**Comment by etienne-lms** [comment]
Review tag applied.
Can be merged since  OP-TEE/manifest#282 has been merged.

**Pull request merged**